File: backend/app/utils/supabase_client.py
from supabase import create_client, Client
import os

SUPABASE_URL = os.getenv("SUPABASE_URL")
SUPABASE_KEY = os.getenv("SUPABASE_KEY")  # service role key

if not SUPABASE_URL:
    raise ValueError("SUPABASE_URL is missing in environment variables")

if not SUPABASE_KEY:
    raise ValueError("SUPABASE_KEY is missing in environment variables")

supabase: Client = create_client(SUPABASE_URL, SUPABASE_KEY)

File: frontend/src/lib/supabaseClient.js
// src/lib/supabaseClient.js
import { createClient } from "@supabase/supabase-js";

const supabaseUrl = import.meta.env.VITE_SUPABASE_URL;
const supabaseAnonKey = import.meta.env.VITE_SUPABASE_ANON_KEY;

if (!supabaseUrl || !supabaseAnonKey) {
  console.warn("Missing Supabase env vars: VITE_SUPABASE_URL or VITE_SUPABASE_ANON_KEY");
}

const SUPABASE_KEY = "supabase_client_singleton_v1";

// attach to window in dev to preserve singleton across HMR
const _global = typeof window !== "undefined" ? window : globalThis;

if (!_global[SUPABASE_KEY]) {
  _global[SUPABASE_KEY] = createClient(supabaseUrl, supabaseAnonKey);
}

export const supabase = _global[SUPABASE_KEY];
export default supabase;
